### DISCUSSION TITLE
Fedora 41 install issue

### DISCUSSION THREAD

**Discussion by mzigaib** [Comfy-Org/ComfyUI-Manager#1245] [Q&A]
I am trying to install ComfyUI-Manager following the instructions but using Python3.12 since 3.13 gives issues to ComfyUI.
And I am getting this error on load.

[START] Security scan
[DONE] Security scan
## ComfyUI-Manager: installing dependencies. (GitPython)
/usr/bin/python3.12: No module named pip
## [ERROR] ComfyUI-Manager: Attempting to reinstall dependencies using an alternative method.
/usr/bin/python3.12: No module named pip
## [ERROR] ComfyUI-Manager: GitPython package seems to be installed, but failed to load somehow. Make sure you have a working git client installed
** ComfyUI startup time: 2024-11-19 00:13:29.506468
** Platform: Linux
** Python version: 3.12.7 (main, Oct  1 2024, 00:00:00) [GCC 14.2.1 20240912 (Red Hat 14.2.1-3)]
** Python executable: /usr/bin/python3.12
** ComfyUI Path: /opt/ComfyUI
** Log path: /opt/ComfyUI/comfyui.log

Prestartup times for custom nodes:
   0.4 seconds: /opt/ComfyUI/custom_nodes/ComfyUI-Manager

Total VRAM 15967 MB, total RAM 128708 MB
pytorch version: 2.5.1+cu124
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 4070 Ti SUPER : cudaMallocAsync
Using pytorch cross attention
[Prompt Server] web root: /opt/ComfyUI/web
Traceback (most recent call last):
  File "/opt/ComfyUI/nodes.py", line 2022, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/opt/ComfyUI/custom_nodes/ComfyUI-Manager/__init__.py", line 6, in <module>
    from .glob import manager_server
  File "/opt/ComfyUI/custom_nodes/ComfyUI-Manager/glob/manager_server.py", line 13, in <module>
    import git
ModuleNotFoundError: No module named 'git'

Cannot import /opt/ComfyUI/custom_nodes/ComfyUI-Manager module for custom nodes: No module named 'git'

Import times for custom nodes:
   0.0 seconds: /opt/ComfyUI/custom_nodes/websocket_image_save.py
   0.0 seconds (IMPORT FAILED): /opt/ComfyUI/custom_nodes/ComfyUI-Manager

Starting server

To see the GUI go to: http://127.0.0.1:8188

I was wondering if someone could help me.
Thanks!

**Comment by ltdrdata** [comment]
You seem to be experiencing many issues with your Python environment. 
First, as I’ve mentioned in other responses, please start by running your project in a venv.

https://github.com/comfyanonymous/ComfyUI/discussions/5668

**Comment by mzigaib** [comment]
That is what I am doing

:~$ cd /opt/ComfyUI/
:/opt/ComfyUI$ python3.12 -m venv venv
:/opt/ComfyUI$ . venv/bin/activate
(venv) :/opt/ComfyUI$ python3.12 ./main.py
[START] Security scan
[DONE] Security scan
## ComfyUI-Manager: installing dependencies done.
** ComfyUI startup time: 2024-11-19 02:25:10.369300
** Platform: Linux
** Python version: 3.12.7 (main, Oct  1 2024, 00:00:00) [GCC 14.2.1 20240912 (Red Hat 14.2.1-3)]
** Python executable: /opt/ComfyUI/venv/bin/python3.12
** ComfyUI Path: /opt/ComfyUI
** Log path: /opt/ComfyUI/comfyui.log

Prestartup times for custom nodes:
   0.8 seconds: /opt/ComfyUI/custom_nodes/ComfyUI-Manager

Traceback (most recent call last):
  File "/opt/ComfyUI/./main.py", line 91, in <module>
    import execution
  File "/opt/ComfyUI/execution.py", line 13, in <module>
    import nodes
  File "/opt/ComfyUI/nodes.py", line 21, in <module>
    import comfy.diffusers_load
  File "/opt/ComfyUI/comfy/diffusers_load.py", line 3, in <module>
    import comfy.sd
  File "/opt/ComfyUI/comfy/sd.py", line 5, in <module>
    from comfy import model_management
  File "/opt/ComfyUI/comfy/model_management.py", line 19, in <module>
    import psutil
ModuleNotFoundError: No module named 'psutil'

**Reply by ltdrdata** [reply]
Once you have created and activated the venv, the Python environment will be empty. Therefore, you need to reinstall the required packages within it.

Please install the r`equirements.txt` files for both ComfyUI and ComfyUI-Manager.

Torch cannot be installed simply by using the requirements file, so make sure to follow the installation guide.
https://github.com/comfyanonymous/ComfyUI?tab=readme-ov-file#manual-install-windows-linux

**Reply by mzigaib** [reply]
Ok got it, let me try it right away.

**Comment by mzigaib** [comment]
Oh now I think it is working, I think what I was missing was to install requirements and torch on both to get it working.

The result:

python main.py 
[START] Security scan
[DONE] Security scan
## ComfyUI-Manager: installing dependencies done.
** ComfyUI startup time: 2024-11-19 12:59:38.933736
** Platform: Linux
** Python version: 3.12.7 (main, Oct  1 2024, 00:00:00) [GCC 14.2.1 20240912 (Red Hat 14.2.1-3)]
** Python executable: /usr/bin/python3.12
** ComfyUI Path: /home/mzigaib/Downloads/ComfyUI/ComfyUI
** Log path: /home/mzigaib/Downloads/ComfyUI/ComfyUI/comfyui.log

Prestartup times for custom nodes:
   0.4 seconds: /home/mzigaib/Downloads/ComfyUI/ComfyUI/custom_nodes/ComfyUI-Manager

Total VRAM 15967 MB, total RAM 128708 MB
pytorch version: 2.5.1+cu124
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 4070 Ti SUPER : cudaMallocAsync
Using pytorch cross attention
[Prompt Server] web root: /home/mzigaib/Downloads/ComfyUI/ComfyUI/web
### Loading: ComfyUI-Manager (V2.51.9)
### ComfyUI Revision: 2837 [9cc90ee3] | Released on '2024-11-18'

Import times for custom nodes:
   0.0 seconds: /home/mzigaib/Downloads/ComfyUI/ComfyUI/custom_nodes/websocket_image_save.py
   0.0 seconds: /home/mzigaib/Downloads/ComfyUI/ComfyUI/custom_nodes/ComfyUI-Manager

Starting server

To see the GUI go to: http://127.0.0.1:8188
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/model-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/github-stats.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json
FETCH DATA from: /home/mzigaib/Downloads/ComfyUI/ComfyUI/custom_nodes/ComfyUI-Manager/extension-node-map.json [DONE]

![image](https://github.com/user-attachments/assets/27ec86ee-b78b-40e2-927c-b6600863d326)

![image](https://github.com/user-attachments/assets/41c5f2f4-e0ba-4689-bdf2-b6d400ebb6a3)

**Comment by mzigaib** [comment]
I think now it is working as it supposed to I was just expecting a floating panel instead of accessing it on top.

Anyway thank you so much for the patience and for the support.
